plaidml2/op/op.h

```diff
@@ -75,6 +75,23 @@ inline edsl::Tensor mean(const edsl::Tensor& I, const edsl::Value& axes = edsl::
   return details::op("mean", args).as_tensor();
 }
 
+inline edsl::Tensor prod(const edsl::Tensor& I, const edsl::Value& axes = edsl::None(), bool keepdims = false) {
+  auto args = edsl::make_tuple(I, axes, keepdims);
+  return details::op("prod", args).as_tensor();
+}
+
+inline edsl::Tensor relu(const edsl::Tensor& I, const edsl::Tensor& alpha = edsl::Placeholder(PLAIDML_DATA_FLOAT32, {}),
```

Comment:
I think we want the default values for `alpha` and `max_value` to be `edsl::None()`

Reply:
Since the op itself expects a tensor for `alpha` and `max_value`, we get these c++ errors when we try to set the default values to `edsl::None`:
```
error: no viable conversion from 'plaidml::edsl::Value' to 'const edsl::Tensor'
```

Reply:
Change the datatype of the argument to be an `edsl::Value`. A `Value` is sort of like a variant, a way to allow multiple data types to be packed into a single type.

Reply:
Actually.
This is a case where in C++ we probably should have multiple versions of this function. One which takes all parameters for `alpha` and `max_value`. Another that only takes `alpha`, another that only takes `max_value` and yet another that doesn't take any.

Then, each impl would pass `edsl::None` to the underlying `args` Value.

Reply:
To be clear, here are the functions:

```
edsl::Tensor relu(const edsl::Tensor& I, float threshold = 0.0);
edsl::Tensor relu(const edsl::Tensor& I, const edsl::Tensor& alpha, float threshold = 0.0);
edsl::Tensor relu(const edsl::Tensor& I, const edsl::Tensor& alpha, const edsl::Tensor& max_value, , float threshold = 0.0);
```

Reply:
Didn't see your last two comments before making my last commit. I'll do this instead.